core/system/src/page_allocator.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if !(defined(UVISOR_PRESENT) && (UVISOR_PRESENT == 1))
+#include "uvisor-lib/uvisor-lib.h"
+
+#include <string.h>
+
+#include <stdio.h>
+#define DPRINTF(...) {}
+#define g_active_box 0
+#define vmpu_is_box_id_valid(...) 0
+#define HALT_ERROR(id, ...) DPRINTF(__VA_ARGS__)
+#else
+#include <uvisor.h>
+#include "page_allocator.h"
+#endif
+
+/* By default a maximum of 32 pages are allowed */
+#ifndef UVISOR_PAGE_TABLE_MAX_COUNT
+#   define UVISOR_PAGE_TABLE_MAX_COUNT ((size_t)32)
+#endif
+/* The page box_id is the box id which is 8bit large  */
+typedef uint8_t page_owner_t;
+/* Maps the page to the owning box handle */
+static page_owner_t page_owner_table[UVISOR_PAGE_TABLE_MAX_COUNT];
+/* define a unused value for the page table */
+#define UVISOR_PAGE_UNUSED ((page_owner_t)(-1))
+
+/* contains the configured page size */
+static uint32_t page_size;
```

Comment:
Please prepend `g_` to all global variables (even static ones), it improves readability.